runtime/runtime.go

```diff
@@ -2145,3 +2184,207 @@ func NewBlockValue(block Block) interpreter.BlockValue {
 		Timestamp: timestampValue,
 	}
 }
+
+func (r *interpreterRuntime) newAccountKeysAddFunction(
+	addressValue interpreter.AddressValue,
+	runtimeInterface Interface,
+) interpreter.HostFunctionValue {
+	return interpreter.NewHostFunctionValue(
+		func(invocation interpreter.Invocation) trampoline.Trampoline {
+			publicKeyValue := invocation.Arguments[0].(*interpreter.CompositeValue)
+			if publicKeyValue.QualifiedIdentifier != sema.PublicKeyTypeName {
```

Comment:
I think we can remove this check here and for the other arguments below, because the static check will already ensure that the value has the correct type.

Reply:
Hmm.. added this check because I figured we could actually send corrupted data from outside (by importing as script/function arguments). Because we are not doing a dynamic type checking for imports. 

But yeah, this is not the place to handle that. I will create a separate issue for that and tackle it separately.

Reply:
Created https://github.com/onflow/cadence/issues/689